### ISSUE TITLE
Uncaught TypeError: target.getModel is not a function

### ISSUE THREAD

**Issue by djputta** [atom-haskell/haskell-ghc-mod#176]
**Atom Version**: 1.10.2
**Electron Version**: 0.37.8
**System**: Ubuntu 16.04.1
**Thrown From**: [haskell-ghc-mod](https://github.com/atom-haskell/haskell-ghc-mod) package, v1.17.2

### Stack Trace

Uncaught TypeError: target.getModel is not a function

```
At /home/dheeraj/.atom/packages/haskell-ghc-mod/lib/upi-consumer.coffee:182

TypeError: target.getModel is not a function
    at UPIConsumer.module.exports.UPIConsumer.insertImportCommand (/home/dheeraj/.atom/packages/haskell-ghc-mod/lib/upi-consumer.coffee:182:21)
    at atom-text-editor.<anonymous> (/home/dheeraj/.atom/packages/haskell-ghc-mod/lib/upi-consumer.coffee:1:1)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/usr/share/atom/resources/app.asar/src/command-registry.js:260:29)
    at CommandRegistry.handleCommandEvent (/usr/share/atom/resources/app.asar/src/command-registry.js:3:61)
    at CommandRegistry.module.exports.CommandRegistry.dispatch (/usr/share/atom/resources/app.asar/src/command-registry.js:160:19)
    at AtomEnvironment.module.exports.AtomEnvironment.dispatchContextMenuCommand (/usr/share/atom/resources/app.asar/src/atom-environment.js:1115:28)
    at EventEmitter.outerCallback (/usr/share/atom/resources/app.asar/src/application-delegate.js:342:25)
    at emitThree (events.js:110:13)
    at EventEmitter.emit (events.js:188:7)
```

### Commands

```
     -3:46 haskell-ghc-mod:check-file (input.hidden-input)
     -2:48.2.0 atom-beautify:beautify-editor (input.hidden-input)
     -2:39.3.0 window:increase-font-size (input.hidden-input)
     -2:39.1.0 settings-view:open (input.hidden-input)
     -0:05.7.0 core:save (input.hidden-input)
     -0:00.0 haskell-ghc-mod:insert-import (div.scroll-view)
```

### Config

```json
{
  "core": {
    "ignoredNames": [
      ".git",
      ".hg",
      ".svn",
      ".DS_Store",
      "._*",
      "Thumbs.db",
      ".exe"
    ],
    "themes": [
      "atom-material-ui",
      "atom-material-syntax"
    ]
  },
  "haskell-ghc-mod": {
    "cabalSandbox": false
  }
}
```

### Installed Packages

```coffee
# User
atom-beautify, v0.29.12 (active)
atom-material-syntax, v0.4.6 (active)
atom-material-syntax-dark, v0.2.5 (inactive)
atom-material-syntax-light, v0.4.5 (inactive)
atom-material-ui, v1.3.6 (active)
atom-yeoman, v0.3.15 (active)
autocomplete-clang, v0.10.0 (active)
autocomplete-go, v1.1.1 (active)
autocomplete-haskell, v0.6.7 (active)
autocomplete-python, v1.8.6 (active)
builder-go, v1.0.1 (active)
file-icons, v1.7.19 (active)
fonts, v1.6.0 (active)
go-config, v1.2.3 (active)
go-debug, v0.1.22 (active)
go-get, v1.0.4 (active)
go-plus, v4.2.0 (active)
godoc, v0.4.0 (active)
gofmt, v1.1.8 (active)
gometalinter-linter, v1.1.1 (active)
gorename, v1.0.3 (active)
haskell-ghc-mod, v1.17.2 (active)
ide-haskell, v1.8.1 (active)
language-haskell, v1.9.9 (active)
language-scala, v1.1.6 (active)
linter, v1.11.16 (active)
linter-clang, v3.4.4 (active)
linter-flake8, v1.13.4 (active)
linter-scalac, v1.4.2 (active)
minimap, v4.25.0 (active)
minimap-bookmarks, v0.4.2 (active)
minimap-cursorline, v0.2.0 (active)
minimap-find-and-replace, v4.5.1 (active)
minimap-git-diff, v4.3.1 (active)
minimap-highlight-selected, v4.4.0 (active)
minimap-selection, v4.4.0 (active)
navigator-godef, v1.0.4 (active)
script, v3.9.0 (inactive)
tester-go, v1.0.6 (active)
atom-dark-syntax, v0.27.0 (inactive)
atom-dark-ui, v0.52.0 (inactive)
atom-light-syntax, v0.28.0 (inactive)
atom-light-ui, v0.44.0 (inactive)
base16-tomorrow-dark-theme, v1.1.0 (inactive)
base16-tomorrow-light-theme, v1.1.1 (inactive)
one-dark-ui, v1.5.0 (inactive)
one-light-ui, v1.5.0 (inactive)
one-dark-syntax, v1.3.0 (inactive)
one-light-syntax, v1.3.0 (inactive)
solarized-dark-syntax, v1.0.2 (inactive)
solarized-light-syntax, v1.0.2 (inactive)
about, v1.5.3 (active)
archive-view, v0.61.1 (active)
autocomplete-atom-api, v0.10.0 (active)
autocomplete-css, v0.11.2 (active)
autocomplete-html, v0.7.2 (active)
autocomplete-plus, v2.31.1 (active)
autocomplete-snippets, v1.11.0 (active)
autoflow, v0.27.0 (inactive)
autosave, v0.23.1 (active)
background-tips, v0.26.1 (active)
bookmarks, v0.41.0 (active)
bracket-matcher, v0.82.1 (active)
command-palette, v0.38.0 (inactive)
deprecation-cop, v0.54.1 (active)
dev-live-reload, v0.47.0 (active)
encoding-selector, v0.22.0 (active)
exception-reporting, v0.39.0 (active)
find-and-replace, v0.201.0 (inactive)
fuzzy-finder, v1.3.0 (active)
git-diff, v1.1.0 (active)
go-to-line, v0.31.0 (inactive)
grammar-selector, v0.48.1 (active)
image-view, v0.58.2 (active)
incompatible-packages, v0.26.1 (active)
keybinding-resolver, v0.35.0 (active)
line-ending-selector, v0.5.0 (active)
link, v0.31.1 (inactive)
markdown-preview, v0.158.0 (active)
metrics, v0.53.1 (active)
notifications, v0.65.0 (active)
open-on-github, v1.2.0 (inactive)
package-generator, v1.0.0 (inactive)
settings-view, v0.242.2 (active)
snippets, v1.0.2 (active)
spell-check, v0.67.1 (active)
status-bar, v1.4.1 (active)
styleguide, v0.47.0 (active)
symbols-view, v0.113.0 (inactive)
tabs, v0.100.2 (active)
timecop, v0.33.2 (active)
tree-view, v0.208.2 (active)
update-package-dependencies, v0.10.0 (active)
welcome, v0.34.0 (active)
whitespace, v0.33.0 (active)
wrap-guide, v0.38.1 (active)
language-c, v0.52.1 (active)
language-clojure, v0.21.0 (active)
language-coffee-script, v0.47.2 (active)
language-csharp, v0.12.1 (active)
language-css, v0.37.1 (active)
language-gfm, v0.88.0 (active)
language-git, v0.15.0 (active)
language-go, v0.42.1 (active)
language-html, v0.45.1 (active)
language-hyperlink, v0.16.0 (active)
language-java, v0.23.0 (active)
language-javascript, v0.119.0 (active)
language-json, v0.18.2 (active)
language-less, v0.29.5 (active)
language-make, v0.22.2 (active)
language-mustache, v0.13.0 (active)
language-objective-c, v0.15.1 (active)
language-perl, v0.35.0 (active)
language-php, v0.37.2 (active)
language-property-list, v0.8.0 (active)
language-python, v0.45.0 (active)
language-ruby, v0.69.0 (active)
language-ruby-on-rails, v0.25.0 (active)
language-sass, v0.56.0 (active)
language-shellscript, v0.22.4 (active)
language-source, v0.9.0 (active)
language-sql, v0.23.0 (active)
language-text, v0.7.1 (active)
language-todo, v0.28.0 (active)
language-toml, v0.18.0 (active)
language-xml, v0.34.9 (active)
language-yaml, v0.26.0 (active)

# Dev
No dev packages
```

**Comment by ron-wolf** [comment]
Why closed?

**Comment by lierdakil** [comment]
This is a rather weird heisenbug, and I'm couldn't reproduce it for the life of me. I should also point out that it was closed by OP.

Try restarting Atom. If it doesn't help, try reinstalling haskell-ghc-mod. If it also doesn't help, we'll try to figure it out. Sorry, I'm really pressed for time right now, so can't really dive into this rabbit hole straight away.

**Comment by ron-wolf** [comment]
It still happens after restarting and reinstalling. It’s not a big deal for me personally (I have other tools I can switch to), but I seem to be reproducing it fairly consistently, so it might be worth trying to fix.

**Comment by lierdakil** [comment]
Okay, so I was able to reproduce with atom 1.13.0-beta8. Either I've had a very lucky break, or it got more pronounced than in 1.12. Apparently event target gets messed up if you don't activate the editor before bringing up the context menu. (Tentative) fix is released as v1.19.3. Please let me know if it works for you. Thanks.
